### PR TITLE
fix: handle reconnection logic for zero queue consumer

### LINKED ISSUE #1330: Zero queue consumer doesn't seem to work properly
#### Expected behavior

omitted

#### Actual behavior

Zero queue consumer has been supported since v0.13.0, but there seem to be some buggy behaviors.

- Immediately after creating a consumer, I got the following error log:
```
ERRO[0000] unable to send initial permits to broker      consumerID=1 error="invalid number of permits requested: 0" name=dufbb subscription=sub1 topic="persistent://pulsar/test/t1"
```
- If I registered `MessageChannel` when creating a consumer, its `availablePermits` was 0. Naturally, even if messages were published to the topic, it was not able to receive any of them.
- Consumer that receive messages using `Receive()` rather than `MessageChannel` worked. However, if the connected topic was unloaded or the broker was restarted, `availablePermits` became 0 and no messages could be received thereafter.

#### Steps to reproduce

I ran the following code:

consumer_with_message_channel.go
```go
func main() {
    client, err := pulsar.NewClient(pulsar.ClientOptions{
        URL: "pulsar://localhost:6650",
    })
    if err != nil {
        log.Fatal(err)
    }
    defer client.Close()

    channel := make(chan pulsar.ConsumerMessage)

    consumer, err := client.Subscribe(pulsar.ConsumerOptions{
        Topic:                   "persistent://pulsar/test/t1",
        SubscriptionName:        "sub1",
        ReceiverQueueSize:       0,
        EnableZeroQueueConsumer: true,
        MessageChannel:          channel,
    })
    if err != nil {
        log.Fatal(err)
    }
    defer consumer.Close()

    for out := range channel {
        msg := out.Message
        fmt.Printf("Received: %s (%s)\n", string(msg.Payload()), msg.ID().String())
        consumer.Ack(msg)
    }
}
```

consumer_without_message_channel.go
```go
func main() {
    client, err := pulsar.NewClient(pulsar.ClientOptions{
        URL: "pulsar://localhost:6650",
    })
    if err != nil {
        log.Fatal(err)
    }
    defer client.Close()

    consumer, err := client.Subscribe(pulsar.ConsumerOptions{
        Topic:                   "persistent://pulsar/test/t1",
        SubscriptionName:        "sub1",
        ReceiverQueueSize:       0,
        EnableZeroQueueConsumer: true,
    })
    if err != nil {
        log.Fatal(err)
    }
    defer consumer.Close()

    for i := 0; i < 100; i++ {
        msg, err := consumer.Receive(context.Background())
        if err == nil {
            fmt.Printf("Received: %s (%s)\n", string(msg.Payload()), msg.ID().String())
            consumer.Ack(msg)
        } else {
            log.Fatal(err)
        }
    }
}
```

#### System configuration
**Pulsar version**: v0.14.0

### PR DESCRIPTION
**Pull request by crossoverJie**
Fixes #1330 #398

### Motivation

`ZeroQueueConsumer` cannot consume messages when the broker restarts or unload topic.

### Modifications
- Increase `availablePermits` on reconnect for `ZeroQueueConsumer` 
- Add test case for zero queue consumer reconnection
- Suppress/Do not print the "unable to send initial permits to broker" error log when using `ZeroQueueConsumer`.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / GoDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar/consumer_zero_queue_test.go [hunk 1]
```diff
@@ -115,6 +121,111 @@ func TestNormalZeroQueueConsumer(t *testing.T) {
...
+	}
+
+	go func() {
+		time.Sleep(3 * time.Second)
+		log.Println("Simulating broker restart...")
+		// Simulate a broker restart by stopping the pulsar container
+		time := 5 * time.Second
+		err = c.Stop(context.Background(), &time)
+		assert.Nil(t, err)
+		err = c.Start(context.Background())
```
Comment (BewareMyPower) [inline pulsar/consumer_zero_queue_test.go]: Could this be replaced by unloading the namespace? Then we don't have to restart the container

File: pulsar/consumer_zero_queue_test.go [hunk 2]
```diff
@@ -115,6 +121,111 @@ func TestNormalZeroQueueConsumer(t *testing.T) {
...
+		expectProperties := map[string]string{
+			"key-1": "pulsar-1",
+		}
+		assert.Equal(t, []byte(expectMsg), msg.Payload())
+		assert.Equal(t, "pulsar", msg.Key())
+		assert.Equal(t, expectProperties, msg.Properties())
+		// ack message
+		consumer.Ack(msg)
+		log.Printf("receive message: %s", msg.ID().String())
+		time.Sleep(5 * time.Second)
```
Comment (BewareMyPower) [inline pulsar/consumer_zero_queue_test.go]: This will sleep for about 1 minute. Is it possible to speed up the tests?

File: pulsar/consumer_zero_queue_test.go (hunk 1 above)
Comment (crossoverJie) [reply pulsar/consumer_zero_queue_test.go]: Done

File: pulsar/consumer_zero_queue_test.go (hunk 2 above)
Comment (crossoverJie) [reply pulsar/consumer_zero_queue_test.go]: I added a `channel`, and the test time was reduced to 10s.

**Review by BewareMyPower — APPROVED** [review]
